toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/popup/views/fragment/PopupFragment.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.test.popup.views.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import androidx.appcompat.app.AlertDialog
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import com.esri.arcgisruntime.mapping.popup.Popup
+import com.esri.arcgisruntime.toolkit.popup.util.observeEvent
+import com.esri.arcgisruntime.toolkit.popup.viewmodel.PopupViewModel
+import com.esri.arcgisruntime.toolkit.test.R
+import com.esri.arcgisruntime.toolkit.test.databinding.FragmentPopupBinding
+import kotlinx.android.synthetic.main.fragment_popup.*
+
+/**
+ * Responsible for displaying a Popup.
+ */
+class PopupFragment : Fragment() {
```

Comment:
For simplicity, could we ommit this fragment and declare all its properties directly in PopupViewTestActivity?

Reply:
No we cannot omit this fragment as we are using it to show the popupView in the bottomsheet. 
The activity contains the MapView and the `BottomsheetContainer`.  The `BottomsheetContainer` is populated with the `PopupFragment`

Reply:
As above for PopupViewTestActivity, some more details of the implementation of this fragment could be explained here in the class level doc, in particular how the binding between PopupViewModel and PopupView is being set up.